src/github.com/getlantern/lantern-android/README.md

```diff
@@ -0,0 +1,232 @@
+# Lantern on Android
+
+```java
+import go.flashlight.Flashlight;
+```
+
+The `lantern-android` repository provides documentation and scripts for
+building a basic [flashlight][1] shared library that exports special methods
+that can be used from Java code, making it possible to run the [flashlight][1]
+client on Android devices.
+
+```java
+try {
+  Flashlight.RunClientProxy("0.0.0.0:9192");
+} catch (Exception e) {
+  throw new RuntimeException(e);
+}
+```
+
+## Prerequisites
+
+* An OSX or Linux box
+* [docker][2]
+* [Android Studio][3]
+* [Go 1.4][4]
+* [GNUMake][6]
+
+### Setting up a development environment
+
+We're going to clone and use the [flashlight-build][5] repository, that
+project provides us with everything we need to build Lantern tools and
+libraries.
+
+```sh
+mkdir -p $GOPATH/src/github.com/getlantern
+cd $GOPATH/src/github.com/getlantern
+git clone https://github.com/getlantern/flashlight-build.git
+```
+
+After cloning the repository use the `setenv.bash` script to export development
+enviromental variables to the current terminal session.
+
+```
+cd flashlight-build
+source setenv.bash
+```
+
+## Building the Android library
+
+`flashlight-build` will provide you with a copy of `lantern-android`, change
+directory into `lantern-android` and execute `make` to start compiling our
+shared library:
+
+```
+cd $GOPATH/src/github.com/getlantern/lantern-android
+
+make
+# ...
+# BUILD SUCCESSFUL
+# Total time: 7 seconds
+```
+
+The `make` command will create a new `app` subdirectory that will contain an
+Android example project. You may import the contents of the `app` subdirectory
+into Android Studio to see libflashlight working.
```

Comment:
@xiam It turns out that I can't run the Android emulator on my Mac if VirtualBox is open.  I forget whether I opened VirtualBox myself or if boot2docker did it during the make process, but it might be nice to add a note about that in here.